service/state/chain.go

```diff
@@ -102,7 +75,7 @@ func (c *Chain) Events(height uint64, types ...string) ([]flow.Event, error) {
 			// Unmarshal event batch and append them to result slice.
 			var evts []flow.Event
 			err := it.Item().Value(func(val []byte) error {
-				val, err := c.core.decompressor.DecodeAll(val, nil)
+				val, err := decoder.DecodeAll(val, nil)
```

Comment:
Hey @Ullaakut, so this still exposes the logic of decoding in the `state` package. There is really no need to have the `state` package aware of anything related to encoding or decoding. Just like the compressing/decompressing, this should be hidden away within the new `storage` package, *including* anything related to key creation.

For this example, for instance, you would only have

```golang
_ = StoreEvents(height, type, events)(tx)
_ = RetrieveEvents(height, types, &events)(tx)
```